### PR TITLE
chore: convert URLShortLinkButton to typescript

### PR DESCRIPTION
**Pull request by ktmud**
### SUMMARY

Convert URLShortLinkButton to TypeScript (tsx) in preparation for bigger refactor where we support including `activeTabs` in dashboard urls (currently you can only highlight one tab using `location.hash`).

No visible changes in the UI.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/URLShortLinkButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  anchorLinkId,
+  emailContent = '',
+  emailSubject = '',
+  placement = 'left',
+}: URLShortLinkButtonProps) {
+  const [shortUrl, setShortUrl] = useState('');
+  const { addDangerToast } = useToasts();
+
+  const getCopyUrl = async () => {
+    if (dashboardId) {
```
Comment (villebro) [inline superset-frontend/src/components/URLShortLinkButton/index.tsx:47]: This got me wondering, can we really have this component without `dashboardId` being populated: And based on a quick search, this appears to be the only component that's using this, and there `dashboardId` is always set: https://github.com/apache/superset/blob/902ac053722ada89f817156a0af38ec03f27376c/superset-frontend/src/components/AnchorLink/index.jsx#L28 So I think we can clean this up further by making `dashboardId` a mandatory prop in `URLShortLinkButtonProps`.

**Review by villebro — APPROVED** [review]
Looks much nicer now, thanks for cleaning this up! 👍 Minor improvement suggestion, other than that LGTM.

File: superset-frontend/src/dashboard/components/gridComponents/Header.jsx [hunk 2]
```diff
@@ -176,11 +176,7 @@ class Header extends React.PureComponent {
                   showTooltip={false}
                 />
                 {!editMode && (
-                  <AnchorLink
-                    anchorLinkId={component.id}
-                    filters={filters}
-                    showShortLinkButton
-                  />
+                  <AnchorLink id={component.id} dashboardId={dashboardId} />
```
Comment (ktmud) [inline superset-frontend/src/dashboard/components/gridComponents/Header.jsx:179]: Adding `dashboardId` fixes the bug where the copy button doesn't copy the right URL for section headers in dashboards.

**Comment by ktmud** [comment]
Merging to unblock followup PRs

**Pull request merged**